public/app/features/provisioning/github.ts

```diff
@@ -0,0 +1,67 @@
+import { Octokit } from 'octokit';
+import { createPullRequest } from 'octokit-plugin-create-pull-request';
+
+import { Resource } from '../apiserver/types';
+
+import { RepositorySpec } from './api/types';
+
+interface PullRequestData {
+  title: string;
+  comment: string;
+  branch: string;
+  path: string;
+  body: object;
+}
+
+export async function createGithubPullRequest(repo: Resource<RepositorySpec>, data: PullRequestData) {
+  const spec = repo.spec.github;
+  if (!spec?.owner || !spec.repository) {
+    return Promise.reject('Repository not configured');
+  }
+
+  const MyOctokit = Octokit.plugin(createPullRequest);
+  const client = new MyOctokit({
+    // TODO!!! can we use oauth web flow?
+    auth: spec.token,
```

Comment:
This path only makes sense if we can do frontend login...

BUT looking at it we need clientId+secret 🤔 
https://github.com/octokit/auth-oauth-user.js/?tab=readme-ov-file#exchange-code-from-oauth-web-flow